File: src/telegram/types/audio.ts

import * as type from '../types'
import Type from './type'


export declare type JsonAudio = {
	file_id: type.String
	file_unique_id: type.String
	file_size?: type.Number
	thumb?: type.JsonPhotoSize
	duration: type.String
	file_name?: type.String
	mime_type?: type.String
	performer?: type.String
	title?: type.String
}

export default class Audio extends Type {
	file_id: type.String
	file_unique_id: type.String
	file_size?: type.Number
	thumb?: type.PhotoSize
	duration: type.String
	file_name?: type.String
	mime_type?: type.String
	performer?: type.String
	title?: type.String

	defaultObject!: JsonAudio

	constructor(audio: JsonAudio) {
		super()

		this.defaultObject = audio

		this.file_id = audio.file_id
		this.file_unique_id = audio.file_unique_id
		this.file_size = audio.file_size
		this.thumb = audio.thumb ? new type.PhotoSize(audio.thumb) : undefined
		this.duration = audio.duration
		this.file_name = audio.file_name
		this.mime_type = audio.mime_type
		this.performer = audio.performer
		this.title = audio.title
	}
}

File: telegram-api-json/pygen/Audio.py

from telelib.bot import DefaultType

# @url https://core.telegram.org/bots/api#audio# ['This object represents an audio file to be treated as music by the Telegram clients.']
class Audio(DefaultType):

    
	# @var file_id Identifier for this file, which can be used to download or reuse the file
	@property
	def file_id(self) -> String:
		return self._d["file_id"]
	# @var file_unique_id Unique identifier for this file, which is supposed to be the same over time and for different bots. Can't be used to download or reuse the file.
	@property
	def file_unique_id(self) -> String:
		return self._d["file_unique_id"]
	# @var duration Duration of the audio in seconds as defined by sender
	@property
	def duration(self) -> Integer:
		return self._d["duration"]
	# @var performer Optional. Performer of the audio as defined by sender or by audio tags
	@property
	def performer(self) -> String:
		return self._d["performer"]
	# @var title Optional. Title of the audio as defined by sender or by audio tags
	@property
	def title(self) -> String:
		return self._d["title"]
	# @var file_name Optional. Original filename as defined by sender
	@property
	def file_name(self) -> String:
		return self._d["file_name"]
	# @var mime_type Optional. MIME type of the file as defined by sender
	@property
	def mime_type(self) -> String:
		return self._d["mime_type"]
	# @var file_size Optional. File size in bytes. It can be bigger than 2^31 and some programming languages may have difficulty/silent defects in interpreting it. But it has at most 52 significant bits, so a signed 64-bit integer or double-precision float type are safe for storing this value.
	@property
	def file_size(self) -> Integer:
		return self._d["file_size"]
	# @var thumb Optional. Thumbnail of the album cover to which the music file belongs
	@property
	def thumb(self) -> PhotoSize:
		return self._d["thumb"]

